packages/push/src/push_vm/stack.rs

```diff
@@ -411,10 +411,10 @@ impl<T> Stack<T> {
     {
         let iter = iter.into_iter();
         // Check that adding these items won't overflow the stack.
-        if !iter
+        if iter
             .len()
             .checked_add(self.size())
-            .is_some_and(|x| x <= self.max_stack_size)
+            .is_none_or(|x| x > self.max_stack_size)
```

Comment:
Nice – that's definitely much better!

Reply:
Yeah that is what that does, they kind of depend on each other, #277 is based on top of this branch so we can either merge them chronologically or we can just merge #277  and then close this as well.